test/e2e/resize_nodes.go

```diff
@@ -230,109 +270,213 @@ var _ = Describe("ResizeNodes", func() {
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("increasing cluster size to %d", replicas+1))
+			err = resizeNodeInstanceGroup(replicas + 1)
+			Expect(err).NotTo(HaveOccurred())
+			err = waitForNodeInstanceGroupSize(replicas + 1)
+			Expect(err).NotTo(HaveOccurred())
+			err = waitForClusterSize(c, replicas+1)
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("increasing size of the replication controller to %d and verifying all pods are running", replicas+1))
+			err = resizeReplicationController(c, ns, name, replicas+1)
+			Expect(err).NotTo(HaveOccurred())
+			pods, err = waitForPodsCreated(c, ns, name, replicas+1)
+			Expect(err).NotTo(HaveOccurred())
+			e = waitForPodsRunning(c, pods)
+			if len(e) > 0 {
+				Failf("Failed to wait for pods running: %v", e)
+			}
+			err = verifyPodsResponding(c, ns, name, pods)
+			Expect(err).NotTo(HaveOccurred())
+		})
 	})
 
-	testName = "should be able to add nodes."
-	It(testName, func() {
-		Logf("starting test %s", testName)
-
-		if !providerIs(supportedProviders...) {
-			By(fmt.Sprintf("Skipping %s test, which is only supported for providers %v (not %s)",
-				testName, supportedProviders, testContext.Provider))
-			return
-		}
-
-		// Create a replication controller for a service that serves its hostname.
-		// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
-		name := "my-hostname-add-node-" + string(util.NewUUID())
-		createServiceWithNameSelector(c, ns, name)
-		replicas := testContext.CloudConfig.NumNodes
-		createServeHostnameReplicationController(c, ns, name, replicas)
-		pods, err := waitForPodsCreated(c, ns, name, replicas)
-		Expect(err).NotTo(HaveOccurred())
-		e := waitForPodsRunning(c, pods)
-		if len(e) > 0 {
-			Failf("Failed to wait for pods running: %v", e)
-		}
-		err = verifyPodsResponding(c, ns, name, pods)
-		Expect(err).NotTo(HaveOccurred())
-
-		By(fmt.Sprintf("increasing cluster size to %d", replicas+1))
-		err = resizeNodeInstanceGroup(replicas + 1)
-		Expect(err).NotTo(HaveOccurred())
-		err = waitForNodeInstanceGroupSize(replicas + 1)
-		Expect(err).NotTo(HaveOccurred())
-		err = waitForClusterSize(c, replicas+1)
-		Expect(err).NotTo(HaveOccurred())
-
-		By(fmt.Sprintf("increasing size of the replication controller to %d and verifying all pods are running", replicas+1))
-		resizeReplicationController(c, ns, name, replicas+1)
-		pods, err = waitForPodsCreated(c, ns, name, replicas+1)
-		Expect(err).NotTo(HaveOccurred())
-		e = waitForPodsRunning(c, pods)
-		if len(e) > 0 {
-			Failf("Failed to wait for pods running: %v", e)
-		}
-		err = verifyPodsResponding(c, ns, name, pods)
-		Expect(err).NotTo(HaveOccurred())
+	Describe("Network", func() {
+		testName = "should survive network partition."
+		It(testName, func() {
+			Logf("starting test %s", testName)
+
+			if !providerIs(supportedProviders...) {
+				By(fmt.Sprintf("Skipping %s test, which is only supported for providers %v (not %s)",
```

Comment:
As above.  You have enough instances of this check it's probably worth factoring it into a utility function.